### DISCUSSION TITLE
"Receive channel has not been made available" when I get request data at global exception

### DISCUSSION THREAD

**Discussion by wuranxu** [fastapi/fastapi#6534] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.exception_handler(Exception)
async def unexpected_exception_error(request: Request, exc: Exception):
    body = await request.body()
    return JSONResponse(
        status_code=status.HTTP_200_OK,
        content=jsonable_encoder({
            "code": 110,
            "request_data": body,
            "msg": str(exc)
        })
    )
```

### Description

Hi guys~I wanna catch global exceptions in my app and return the `body` of request. But now when I write this code above, I got an error:

`RuntimeError: Receive channel has not been made available`

And I checked the error, because request object in `site-packages\starlette\middleware\errors.py` doesn't have `receive` property.

Can you help me ? I have created middleware and diy my Exception, But I can't get exception info any more.(I can get request body) 

### Operating System

Windows

### FastAPI Version

0.63.0

### Python Version

3.9.5

**Comment by Dustyposa** [comment]
Maybe this [issue](https://github.com/tiangolo/fastapi/issues/1216) can help you.

**Comment by wuranxu** [comment]
> Maybe this [issue](https://github.com/tiangolo/fastapi/issues/1216) can help you.

Thanks~ I have solved it by middleware.😂
```python
pity = FastAPI()


async def set_body(request: Request, body: bytes):
    async def receive() -> Message:
        return {"type": "http.request", "body": body}

    request._receive = receive


async def get_body(request: Request) -> bytes:
    body = await request.body()
    await set_body(request, body)
    return body


@pity.middleware("http")
async def errors_handling(request: Request, call_next):
    body = await request.body()
    try:
        await set_body(request, await request.body())
        return await call_next(request)
    except Exception as exc:
        return JSONResponse(
            status_code=status.HTTP_200_OK,
            content=jsonable_encoder({
                "code": 110,
                "msg": str(exc),
                "request_data": body,
            })
        )


```

**Comment by Akkarine** [comment]
@wuranxu , nice workaround! But how do you use `get_body`?
